omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/ConceptController1_8Test.java

```diff
@@ -393,6 +395,50 @@ public void shouldRemoveAnswersFromConcept() throws Exception {
 	}
 
 	@Test
+	public void shouldSetMappingsOnConcept() throws Exception {
+		//before adding
+		Concept concept = service.getConceptByUuid(getUuid());
+		assertThat(concept.getConceptMappings().size(), is(0));
+
+		//add one mapping
+		MockHttpServletRequest request = request(RequestMethod.POST, getURI() + "/" + getUuid());
+		ConceptReferenceTerm referenceTerm = service.getAllConceptReferenceTerms().get(0);
+		String mapTypeUuid = service.getDefaultConceptMapType().getUuid();
+		String json = "{ \"mappings\": [{\"conceptReferenceTerm\":\""+referenceTerm.getUuid()+"\",\"conceptMapType\":\""+mapTypeUuid+"\"}]}";
+		request.setContent(json.getBytes());
+
+		handle(request);
+
+		concept = service.getConceptByUuid(getUuid());
+		assertThat(concept.getConceptMappings().size(), is(1));
+		assertThat(hasMappingToTerm(concept, referenceTerm), is(true));

```

Comment:
For the future... please use available hamcrest matchers or implement FeatureMatcher if not available. Here for you could have used:
{code}
private Matcher<ConceptMap> hasTerm(final ConceptReferenceTerm term) {
   return new FeatureMatcher<ConceptMap, ConceptReferenceTerm >(equalTo(term), "conceptReferenceTerm", "conceptReferenceTerm") {
      @Override
      protected ConceptReferenceTerm featureValueOf(final ConceptMap actual) {
         return actual.getConceptReferenceTerm();
      }
   };
}
....
assertThat(concept.getConceptMappings(), hasItem(hasTerm(referenceTerm)));
{code}

You get a more readable message when there's an error for free. See http://www.planetgeek.ch/2012/03/07/create-your-own-matcher/